src/test/java/org/codebusters/audiogeek/spotifygazer/infrastructure/dataexchange/DatabaseDataModifyAdapterTest.java

```diff
@@ -0,0 +1,87 @@
+package org.codebusters.audiogeek.spotifygazer.infrastructure.dataexchange;
+
+import org.codebusters.audiogeek.spotifygazer.domain.dataexchange.model.AddAlbumCommand;
+import org.codebusters.audiogeek.spotifygazer.domain.newreleasesflow.model.Album;
+import org.codebusters.audiogeek.spotifygazer.domain.newreleasesflow.model.Artist;
+import org.codebusters.audiogeek.spotifygazer.infrastructure.db.repo.AlbumRepository;
+import org.codebusters.audiogeek.spotifygazer.infrastructure.db.repo.ArtistRepository;
+import org.codebusters.audiogeek.spotifygazer.infrastructure.db.repo.GenreRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.net.URI;
+import java.time.LocalDate;
+import java.util.Set;
+
+import static java.time.format.DateTimeFormatter.ofPattern;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest
+@ActiveProfiles("test")
+class DatabaseDataModifyAdapterTest {
+
+    @Autowired
+    private AlbumRepository albumRepo;
+    @Autowired
+    private ArtistRepository artistRepo;
+    @Autowired
+    private GenreRepository genreRepo;
+    @Autowired
+    private DatabaseDataModifyAdapter adapter;
+
+    @BeforeEach
+    void clearDatabase() {
+        albumRepo.deleteAll();
+        artistRepo.deleteAll();
+        genreRepo.deleteAll();
+    }
+
+    @Test
+    @Transactional
```

Comment:
We do not want to use `@Transactional` annotation in test. This will not completely verify your service layer.